controllers/components/modelcontroller/modelcontroller_controller.go

```diff
@@ -0,0 +1,95 @@
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/gc"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render/kustomize"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/security"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+var serviceAccounts = map[cluster.Platform][]string{
+	cluster.SelfManagedRhods: {componentsv1.ModelControllerComponentName},
+	cluster.ManagedRhods:     {componentsv1.ModelControllerComponentName},
+	cluster.OpenDataHub:      {componentsv1.ModelControllerComponentName},
+	cluster.Unknown:          {componentsv1.ModelControllerComponentName},
+}
+
+func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor(
+		mgr,
+		&componentsv1.ModelController{},
+	).
+		// customized Owns() for Component with new predicates
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&promv1.ServiceMonitor{}).
+		Owns(&networkingv1.NetworkPolicy{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.ClusterRole{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&corev1.Service{}).
+		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
+		Owns(&templatev1.Template{}).
+		Owns(&appsv1.Deployment{}, reconciler.WithPredicates(resources.NewDeploymentPredicate())).
+		WatchesGVK(gvk.ModelMeshServing, reconciler.Dynamic()).
+		WatchesGVK(gvk.Kserve, reconciler.Dynamic()).
+		Watches(&extv1.CustomResourceDefinition{}). // call ForLabel() + new predicates
+		// Add ModelController specific actions
+		WithAction(initialize).
+		WithAction(devFlags). // devFlags triggerd by changes in DSC kserve and ModelMeshServing
+		WithAction(security.NewUpdatePodSecurityRoleBindingAction(serviceAccounts)).
+		WithAction(kustomize.NewAction(
+			kustomize.WithCache(),
+			kustomize.WithLabel(labels.ODH.Component(ComponentName), "true"),
+			kustomize.WithLabel(labels.K8SCommon.PartOf, ComponentName),
+		)).
+		WithAction(deploy.NewAction(
+			deploy.WithCache(),
+		)).
+		WithAction(patchOwnerReference).
```

Comment:
I don't think the model controller should change it's ownership, that should be a task for the DSC or Kserve/ModelMesh

Reply:
split the function, to kserve and modelmesh now.